Mono.Debugging/Mono.Debugging.Evaluation/AsyncOperationBase.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Mono.Debugging.Client;
+
+namespace Mono.Debugging.Evaluation
+{
+	public class OperationResult<TValue>
+	{
+		public TValue Result { get; private set; }
+		public bool ResultIsException { get; private set; }
+
+		public OperationResult (TValue result, bool resultIsException)
+		{
+			Result = result;
+			ResultIsException = resultIsException;
+		}
+	}
+
+	public static class OperationResultEx
+	{
+		public static OperationResult<TValue> ThrowIfException<TValue> (this OperationResult<TValue> result, EvaluationContext ctx)
```

Comment:
I guess this approach makes the code more simple. OperationResult is just a data class without any internal logic